backends/tfhe-cuda-backend/cuda/src/zk/expand.cuh

```diff
@@ -0,0 +1,73 @@
+#ifndef EXPAND_CUH
+#define EXPAND_CUH
+
+#include "device.h"
+#include "polynomial/functions.cuh"
+#include "polynomial/polynomial_math.cuh"
+#include "zk/zk.h"
+#include <cstdint>
+
+#include "utils/helper.cuh"
+
+// Expand a LweCompactCiphertextList into a LweCiphertextList
+// - Each x-block computes one output ciphertext
+template <typename Torus, class params>
+__global__ void lwe_expand(Torus const *lwe_compact_array_in,
+                           Torus *lwe_array_out,
+                           const uint32_t *compact_list_length_per_thread,
+                           const uint32_t *lwe_compact_input_indexes) {
+  const auto lwe_output_id = blockIdx.x;
+  const auto lwe_dimension = params::degree;
+
+  const auto compact_list_length =
+      compact_list_length_per_thread[lwe_output_id];
```

Comment:
Maybe we could rename `compact_list_length_per_thread` into `compact_list_length_per_lwe`?